src/window/building/depot.c

```diff
@@ -0,0 +1,441 @@
+#include "depot.h"
+
+#include "building/building.h"
+#include "building/storage.h"
+#include "building/warehouse.h"
+#include "figure/figure.h"
+#include "graphics/generic_button.h"
+#include "graphics/image.h"
+#include "graphics/lang_text.h"
+#include "graphics/panel.h"
+#include "graphics/screen.h"
+#include "graphics/scrollbar.h"
+#include "graphics/text.h"
+#include "graphics/window.h"
+#include "translation/translation.h"
+#include "window/building_info.h"
+
+static void order_set_source(int index, int param2);
+static void order_set_destination(int index, int param2);
+static void order_set_resource(int index, int param2);
+static void order_set_condition_type(int index, int param2);
+static void order_set_condition_threshold(int index, int param2);
+static void set_order_resource(int depot_building_id, int resource_id);
+
+static void setup_for_selected_depot(building_info_context *c);
+
+#define DEPOT_BUTTONS_X_OFFSET 32
+#define DEPOT_BUTTONS_Y_OFFSET 204
+#define ROW_HEIGHT 22
+#define MAX_VISIBLE_ROWS 15
+
+static struct {
+    int focus_button_id;
+    int orders_focus_button_id;
+    int resource_focus_button_id;
+    int storage_building_focus_button_id;
+    int depot_resource_focus_button_id;
+    int depot_building_id;
+    resource_type target_resource_id;
+} data;
+
+static const translation_key TRANSLATION_KEY_ORDER_CONDITION[4] = {
+    TR_ORDER_CONDITION_NEVER,
+    TR_ORDER_CONDITION_ALWAYS,
+    TR_ORDER_CONDITION_SOURCE_HAS_MORE_THAN,
+    TR_ORDER_CONDITION_DESTINATION_HAS_LESS_THAN
+};
```

Comment:
Instead of this, if we have all translation strings next to each other we only need the offset of the first one to find the ones after, avoiding the array,

Reply:
Done.